nearcore/src/migrations.rs

```diff
@@ -4,12 +4,12 @@ use near_primitives::runtime::migration_data::MigrationData;
 use near_primitives::types::Gas;
 use near_primitives::utils::index_to_bytes;
 use near_store::migrations::{set_store_version, BatchedStoreUpdate};
-use near_store::DBCol;
+use near_store::{DBCol, Temperature};
 
 /// Fix an issue with block ordinal (#5761)
 // This migration takes at least 3 hours to complete on mainnet
 pub fn migrate_30_to_31(store_opener: &near_store::StoreOpener, near_config: &crate::NearConfig) {
-    let store = store_opener.open();
+    let store = store_opener.open().get_store(Temperature::Hot);
```

Comment:
I'm a little surprised, that the Temperature parameter is 'leaking' into so many places.

As AFAIK the 'hot' should be the default for almost all the calls (and only few selected ones will be interacting with cold).

Reply:
This is migration code so it’s kinda expected that the code will be a bit more messy and bypass abstractions.  And since the code knows the database is in old version we can just assume that it does not have cold storage.

In general I wanted to make it explicit whenever one accesses storage which storage is accessed.  While many of the call sites will access hot by default, they also won’t have access to NodeStorage but rather to Store.  So for example Runtime will just have Store.  It won’t be able to interact with with storage other than the one it’s assigned to.

The temperature will need to be handled at the entry points though.